### PR TITLE
BAEL-5810: Using Condition Statements in Spring WebFlux

### PR DESCRIPTION
**Pull request by tapankavasthi**

### CODE REVIEW

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        flux = flux.defaultIfEmpty("No words found!");
+
+        StepVerifier.create(flux)
+            .expectNext("No words found!")
+            .verifyComplete();
+    }
+
+    @Test
+    public void givenStringFlux_whenFlatMapped_thenReturnTransformedFlattenedSequence() {
+
```
Comment (theangrydev) [inline spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java:83]: remove whitespace
```suggestion
```

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        Mono<String> noUserAgeCategory = Mono.empty()
+            .zipWhen(id -> Mono.defer(() -> Mono.just(20)), (id, age) -> age >= 18 ? "ADULT" : "KID");
+
+        StepVerifier.create(noUserAgeCategory)
+            .verifyComplete();
+    }
+
+    @Test
+    public void givenTwoDeferredMonos_whenFirstWithValueCalled_thenReturnFirstAvailableMono() {
+
```
Comment (theangrydev) [inline spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java:169]: remove whitespace
```suggestion
```

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        Mono<String[]> mono = Mono.firstWithValue(source1, source2);
+
+        StepVerifier.create(mono)
+            .expectNextMatches(item -> "val".equals(item[0]) && "source2".equals(item[1]))
+            .verifyComplete();
+    }
+
+    @Test
+    public void givenTwoDeferredMonosWithSameValue_whenZippedAndCompared_thenReturnTrueIfEqual() {
+
```
Comment (theangrydev) [inline spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java:185]: remove whitespace
```suggestion
```

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java (hunk 3 above)
Comment (tapankavasthi) [reply spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java]: Fixed

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java (hunk 2 above)
Comment (tapankavasthi) [reply spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java]: Fixed

File: spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java (hunk 1 above)
Comment (tapankavasthi) [reply spring-5-webflux-2/src/test/java/com/baeldung/webflux/condition/ConditionalFlowUnitTest.java]: Fixed

**Review by theangrydev — APPROVED** [review]

**Pull request merged**